### PR TITLE
Create subset_sum.cpp

### PR DESCRIPTION
**Pull request by KillerAV**
#### Description of Change

Writing algorithm for solving subset sum problem using dynamic programming

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

**Review by mishraabhinn — APPROVED** [review]

**Comment by KillerAV** [comment]
@mishraabhinn can you please re-review the PR?

**Review by mishraabhinn — DISMISSED** [review]

**Comment by KillerAV** [comment]
@mishraabhinn can you please help me trigger the workflows?

File: dynamic_programming/subset_sum.cpp [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+ * the array. For example, array = [1, -10, 2, 31, -6], targetSum = -14. 
+ * Output: true => We can pick subset [-10, 2, -6] with sum as 
+ * (-10) + 2 + (-6) = -14.
+ * @author [KillerAV] (https://github.com/KillerAV)
+ */
+
+#include <cassert>    // for std::assert
+#include <iostream>   // for io operations
+#include <vector>   // for vector
+#include <unordered_map>   // for unordered map
```
Comment (Panquesito7) [inline dynamic_programming/subset_sum.cpp:20]:
```suggestion
#include <cassert>        /// for std::assert
#include <iostream>       /// for IO operations
#include <vector>          /// for std::vector
#include <unordered_map>   /// for unordered map
```

File: dynamic_programming/subset_sum.cpp [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+ * whether a subset with target sum exists or not.
+ *
+ * @details
+ * In this problem, we use dynamic programming to find if we can pull out a 
+ * subset from an array whose sum is equal to a given target sum. The overall 
+ * time complexity of the problem is O(n * targetSum) where n is the size of 
+ * the array. For example, array = [1, -10, 2, 31, -6], targetSum = -14. 
+ * Output: true => We can pick subset [-10, 2, -6] with sum as 
+ * (-10) + 2 + (-6) = -14.
+ * @author [KillerAV] (https://github.com/KillerAV)
```
Comment (Panquesito7) [inline dynamic_programming/subset_sum.cpp:14]:
```suggestion
 * @author [KillerAV](https://github.com/KillerAV)
```

**Review by Panquesito7 — DISMISSED** [review]
Awesome work! Thanks. 🚀

**Comment by tjgurwara99** [comment]
Please fix the CI checks that are failing 🙏🏼

**Review by Panquesito7 — APPROVED** [review]

**Review by tjgurwara99 — APPROVED** [review]

**Pull request merged**